src/include/catalog/catalog.h

```diff
@@ -131,7 +133,7 @@ class Catalog {
     // TODO(Wan,chi): This should be refactored into a private ctor for the binder tests, we shouldn't allow nullptr.
     // When bpm_ == nullptr, it means that we're running binder tests (where no txn will be provided). We don't need
     // to create TableHeap in this case.
-    if (bpm_ != nullptr) {
+    if (bpm_ != nullptr && create_table_heap) {
```

Comment:
We can remove the `bpm_ == nullptr` check and the above todo block, since we have the flag now?